charts/opentelemetry-operator/templates/clusterrole.yaml

```diff
@@ -210,6 +224,14 @@ rules:
       - get
       - patch
       - update
+  - apiGroups:
+    - opentelemetry.io
+    resources:
+    - targetallocators/status
+    verbs:
+    - get
+    - patch
+    - update
```

Comment:
What in the new operator release needs these permissions?

Reply:
Those were added in Generate only [TargetAllocator CR from Collector CR #3402](https://github.com/open-telemetry/opentelemetry-operator/pull/3402/files), but looking at it now, these might be required only `targetAllocatorCR` flag is set

Reply:
In that case we should make them conditional in the chart if we can.

Reply:
Agree, the permissions will be set if `.Values.manager.featureGatesMap.operator.targetAllocatorCR` is enabled